### DISCUSSION TITLE
Is there an analog of `postStartCommand` for Features?

### DISCUSSION THREAD

**Discussion by jmuchovej** [devcontainers/community#34] [Q&A] [answered]
I'm building a feature for `asdf`. To avoid creating features for each `asdf-<plugin>`, I was planning on running a script similar to:
```bash
# Find `.tool-versions` within `${WORKSPACE_MOUNT}`, ignoring hidden directories.
/bin/find ${WORKSPACE_MOUNT} -type d -path '*/.*' -prune -o -name '*.tool-version*' -print | while read filePath; do
  echo -e "Setting up \`asdf\` for ${filePath}"

  # Install plugins based on `.tool-versions` contents
  cat $filePath | cut -d' ' -f1 | grep "^[^\#]" | xargs -i asdf plugin add {}

  # Install versions at ${filePath}
  (cd $(dirname ${filePath}) && asdf install)
done
```

Of course, this could be provided to the user and they would need to manually trigger this – but I'm wondering if maybe adding a Feature could "register" a `postStarupCommand` of sorts.

**Comment by bamurtaugh** [comment, marked as answer]
Hi, thanks for your question! I believe this would be covered by this spec proposal to allow Features to contribute lifecycle hooks: https://github.com/devcontainers/spec/issues/181 (cc @joshspicer).
